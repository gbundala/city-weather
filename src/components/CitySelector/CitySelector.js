import React from "react";
import "./CitySelector.css";
import Button from "react-bootstrap/Button";
import FormControl from "react-bootstrap/FormControl";
import InputGroup from "react-bootstrap/InputGroup";

/**
 *
 * The CitySelector is basically as presentational component
 * similar to the WeatherData component. Where as the WeatherData
 * component handles the presentation of the data returned by
 * the API call, this component handles the presentation
 * of the input control and Search button. The props are passed
 * in from the parent component to pass in the data and trigger
 * some state changes
 */
export default function CitySelector({
  setCitySelection,
  citySelection,
  setLoading,
  setSearchCity,
}) {
  return (
    <div className="city-selector-wrapper">
      <InputGroup className="mb-2">
        <FormControl
          placeholder="Search your city..."
          aria-label="Recipient's username"
          aria-describedby="basic-addon2"
          onChange={(e) => {
            setCitySelection(e.target.value);
          }}
          id="cityselector"
          name="cityselector"
          value={citySelection}
        />
        <Button
          variant="outline-secondary"
          id="button-addon2"
          onClick={() => {
            setLoading(true);
            setSearchCity(citySelection);
          }}
        >
          Search
        </Button>
      </InputGroup>
    </div>
  );
}
