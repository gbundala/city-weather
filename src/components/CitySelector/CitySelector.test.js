import React from "react";
import renderer from "react-test-renderer";
import CitySelector from "./CitySelector";

// Snapshot testing

// testing whether the CitySelector presentational component
// renders correctly and matches the saved snapshot

test("renders correctly", () => {
  const tree = renderer.create(<CitySelector />).toJSON();

  expect(tree).toMatchSnapshot();
});
