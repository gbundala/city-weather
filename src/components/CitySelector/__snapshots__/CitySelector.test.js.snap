// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="city-selector-wrapper"
>
  <div
    className="mb-2 input-group"
  >
    <input
      aria-describedby="basic-addon2"
      aria-label="Recipient's username"
      className="form-control"
      id="cityselector"
      name="cityselector"
      onChange={[Function]}
      placeholder="Search your city..."
    />
    <button
      className="btn btn-outline-secondary"
      disabled={false}
      id="button-addon2"
      onClick={[Function]}
      type="button"
    >
      Search
    </button>
  </div>
</div>
`;
